### DISCUSSION TITLE
Labels swapped by yolo

### DISCUSSION THREAD

**Discussion by MateuszGalinski** [ultralytics/ultralytics#15717] [Q&A] [answered]
My issue is as stated in the title, during training my labels got swapped, model correctly finds object but completely misslabels them for example Person is labeled as no-goggles. Dataset was downloaded from roboflow and used as is

this is my data.yaml file
```
train: C:/Users/LocalAdminUser02/Desktop/PPEDetection/dataset/train/images
val: C:/Users/LocalAdminUser02/Desktop/PPEDetection/dataset/valid/images
test: C:/Users/LocalAdminUser02/Desktop/PPEDetection/dataset/test/images

nc: 7
names: ['Person', 'goggles', 'helmet', 'no-goggles', 'no-helmet', 'no-vest', 'vest']

roboflow:
  workspace: ai-camp-safety-equipment-detection
  project: ppe-detection-using-cv
  version: 3
  license: CC BY 4.0
  url: https://universe.roboflow.com/ai-camp-safety-equipment-detection/ppe-detection-using-cv/dataset/3
``` 
example labeled image:
![image_4_jpg rf 7d88b5643e850eb531579672b5aa24db](https://github.com/user-attachments/assets/fbc32324-6b56-4be6-852c-88569de83c4c)
and labels for it: 
```
2 0.3560307692307692 0.19285564853556486 0.23975384615384615 0.3117991631799163
1 0.36552307692307695 0.3049581589958159 0.13053846153846152 0.10712343096234309
6 0.24475384615384616 0.6526150627615063 0.3556692307692308 0.5130334728033472
0 0.35052307692307694 0.47444560669456065 0.5712692307692308 0.8710355648535565
```
I train model like this results = model.train(data = os.path.join("dataset_goggles", "data.yaml"), epochs = 100) and try to load best.pt best_model = YOLO(os.path.join("runs", "detect", "train", "weights", "best.pt")) and plot images as follows: 
```
for result in results:
    img = result.plot()

img_rgb = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)

plt.imshow(img_rgb)
plt.axis('off')  # Hide axes
plt.show()
``` 
and results I get are following 
![image](https://github.com/user-attachments/assets/540a28d6-2871-49a2-b369-92c76a23a186)
Any suggestions, only thing I found about swapped labels is if you use spaces in classes names but i dont use any?

**Comment by glenn-jocher** [comment, marked as answer]
@MateuszGalinski it appears that your labels might be mismatched due to an issue with the dataset or the training process. Please ensure that the class order in your `data.yaml` file matches the order in your dataset's annotations. Additionally, verify that the issue persists with the latest version of the Ultralytics package. If the problem continues, consider re-downloading the dataset from Roboflow to ensure data integrity.

**Comment by MateuszGalinski** [comment]
It turns out I had a typo in .yaml file and used a dataset from different path. Thank you for your help!

**Reply by glenn-jocher** [reply]
@MateuszGalinski i'm glad to hear you resolved the issue! If you encounter any more problems or have further questions, feel free to reach out. Happy training!
